src/views/payment-sheet-views/card-view.js

```diff
@@ -614,32 +609,26 @@ CardView.prototype.onSelection = function () {
   }
 };
 
-CardView.prototype._hideUnsupportedCardIcons = function () {
-  var supportedCardTypes = this.client.getConfiguration().gatewayConfiguration.creditCards.supportedCardTypes;
-
-  Object.keys(constants.configurationCardTypes).forEach(function (paymentMethodCardType) {
-    var cardIcon;
-    var configurationCardType = constants.configurationCardTypes[paymentMethodCardType];
-
-    if (supportedCardTypes.indexOf(configurationCardType) === -1) {
-      cardIcon = this.getElementById(paymentMethodCardType + '-card-icon');
-      classList.add(cardIcon, 'braintree-hidden');
-    }
-  }.bind(this));
-};
+CardView.prototype._showSupportedCardIcons = function () {
+  var self = this;
 
-CardView.prototype._isCardTypeSupported = function (cardType) {
-  var configurationCardType = constants.configurationCardTypes[cardType];
-  var supportedCardTypes = this.client.getConfiguration().gatewayConfiguration.creditCards.supportedCardTypes;
+  this.hostedFieldsInstance.getSupportedCardTypes().then(function (supportedCardTypes) {
+    Object.keys(constants.configurationCardTypes).forEach(function (paymentMethodCardType) {
+      var cardIcon;
+      var configurationKey = constants.configurationCardTypes[paymentMethodCardType];
 
-  return supportedCardTypes.indexOf(configurationCardType) !== -1;
+      if (supportedCardTypes.indexOf(paymentMethodCardType) > -1) {
+        cardIcon = self.getElementById(configurationKey + '-card-icon');
+        classList.remove(cardIcon, 'braintree-hidden');
```

Comment:
It may make sense to either extract these two lines to a different method, or change the method name to show that this method modifies the DOM.

Reply:
Ignore pls. Realize now the new method name does indicate dom action.